app/services/supplier_chooser.rb

```diff
@@ -1,16 +1,86 @@
+  # | Supplier2            | Supplier1           | Supplier2       |
+  # | Supplier2            | Supplier2           | Supplier2       |
+
+  # UPDATING EXISTING RECORDS
+  # | Location+Date Owner | Existing Owner | Doorkeeper App Owner | Resulting Owner |
+  # |---------------------|----------------|----------------------|-----------------|
+  # | nil                 | nil            | nil                  | nil             |
+  # | nil                 | nil            | Supplier1            | Supplier1       |
+  # | nil                 | nil            | Supplier2            | Supplier2       |
+  #
+  # | nil                 | Supplier1      | nil                  | Supplier1       |
+  # | nil                 | Supplier1      | Supplier1            | Supplier1       |
+  # | nil                 | Supplier1      | Supplier2            | Supplier1       |
+  #
+  # | nil                 | Supplier2      | nil                  | Supplier2       |
+  # | nil                 | Supplier2      | Supplier1            | Supplier2       |
+  # | nil                 | Supplier2      | Supplier2            | Supplier2       |
+  #
+  # | Supplier1           | nil            | nil                  | Supplier1       |
+  # | Supplier1           | nil            | Supplier1            | Supplier1       |
+  # | Supplier1           | nil            | Supplier2            | Supplier1       |
+  #
+  # | Supplier1           | Supplier1      | nil                  | Supplier1       |
+  # | Supplier1           | Supplier1      | Supplier1            | Supplier1       |
+  # | Supplier1           | Supplier1      | Supplier2            | Supplier1       |
+  #
+  # | Supplier1           | Supplier2      | nil                  | Supplier1       |
+  # | Supplier1           | Supplier2      | Supplier1            | Supplier1       |
+  # | Supplier1           | Supplier2      | Supplier2            | Supplier1       |
+  #
+  # | Supplier2           | nil            | nil                  | Supplier2       |
+  # | Supplier2           | nil            | Supplier1            | Supplier2       |
+  # | Supplier2           | nil            | Supplier2            | Supplier2       |
+  #
+  # | Supplier2           | Supplier1      | nil                  | Supplier2       |
+  # | Supplier2           | Supplier1      | Supplier1            | Supplier2       |
+  # | Supplier2           | Supplier1      | Supplier2            | Supplier2       |
+  #
+  # | Supplier2           | Supplier2      | nil                  | Supplier2       |
+  # | Supplier2           | Supplier2      | Supplier1            | Supplier2       |
+  # | Supplier2           | Supplier2      | Supplier2            | Supplier2       |
+
   def call
+    if new_record
+      doorkeeper_application_owner || effective_supplier_for_location
+    else
+      effective_supplier_for_location || existing_owner || doorkeeper_application_owner
```

Comment:
I don't follow why this would need to consider `doorkeeper_application_owner` - as `existing_owner` should always be present? Could this be simplified to:

```ruby
effective_supplier_for_location || existing_owner
```

If so then that means we only need to worry about the doorkeeper token when creating the move, which happens in fewer places so there's less need to pass the doorkeeper token around which should simplify the PR.